### PR TITLE
 Update text background settings

### LINKED ISSUE #1849: [FEATURE] Allow setting paint effects on label background shapes
Original commit: https://github.com/qgis/QGIS/commit/debe1098cbff5a8ebfa99491f8c74e30178a7de1 by nyalldawson

Can also be useful for improving text readability through
outer glows and blurs

### PR DESCRIPTION
**Pull request by DelazJ**
Following the same structure as previous tabs, with a screenshot
Fixes #1849, refs #3899

### CODE REVIEW

**Comment by lajunek** [comment]
@DelazJ do we have to Merge the pull request?

**Comment by DelazJ** [comment]
@lajunek after it's reviewed...

File: source/docs/user_manual/style_library/label_settings.rst [hunk 1]
```diff
@@ -67,9 +67,7 @@ In the |text| :guilabel:`Text` tab, you can set:
 * the :guilabel:`Font`, from the ones available on your machine
 * the :guilabel:`Style`: along with the common styles of the font, you can set
   whether the text should be underlined or striked through
-* the :guilabel:`Size` that can be in :guilabel:`Points`, :guilabel:`Pixels`,
-  :guilabel:`Millimeters`, :guilabel:`Map units`, :guilabel:`Meters at scale`,
-  or :guilabel:`Inches`
+* the :guilabel:`Size` in any :ref:`supported units <unit_selector>`
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
* the :guilabel:`Size` in any :ref:`supported unit <unit_selector>`
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 2]
```diff
@@ -161,9 +159,7 @@ Buffer tab
 To create a buffer around the label, activate the |checkbox| :guilabel:`Draw
 text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
 
-* Set the buffer's :guilabel:`Size` that can be in :guilabel:`Points`,
-  :guilabel:`Pixels`, :guilabel:`Millimeters`, :guilabel:`Map units`,
-  :guilabel:`Meters at scale` or :guilabel:`Inches`
+* Set the buffer's :guilabel:`Size` in any :ref:`supported units <unit_selector>`
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
* Set the buffer's :guilabel:`Size` in any :ref:`supported unit <unit_selector>`
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 3]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
-and :guilabel:`Size Y`. You can set a :guilabel:`Rotation` where you can choose
-between 'Sync with label', 'Offset of label' and 'Fixed'.
-Using 'Offset of label' and 'Fixed', you can rotate the background.
-Define an :guilabel:`Offset X,Y` with X and Y values, and the background
-will be shifted. When applying :guilabel:`Radius X,Y`, the background gets
-rounded corners.
-Again, it is possible to mix the background with the underlying layers in the
-map canvas using the :guilabel:`Blend mode` (see :ref:`blend-modes`).
+The |labelbackground| :guilabel:`Background` tab allows you to configure a
+shape that stays at the bottom of the labels. To add a background, activate
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
shape that stays below each label. To add a background, activate
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 4]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+the |unchecked| :guilabel:`Draw Background` checkbox and select
+the :guilabel:`Shape` type. It can be:
+
+* of a regular type such as :guilabel:`Rectangle`, :guilabel:`Square`,
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
* a regular shape such as :guilabel:`Rectangle`, :guilabel:`Square`,
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 5]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+  :guilabel:`Circle` or :guilabel:`Ellipse`
+* an :guilabel:`SVG` symbol from a file, a URL or embedded in the project
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
* an :guilabel:`SVG` symbol (from a file or a URL, embedded in the project
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 6]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+  or style database (:ref:`more details <svg_paths>`)
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
  or style database - :ref:`more details <svg_paths>`)
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 7]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+  :ref:`symbol library <vector_marker_symbols>`.
+
+.. _figure_textsettings_background:
+
+.. figure:: img/textsettings_background.png
+   :align: center
+
+   Text settings - Background tab
+
+Depending on the selected shape, you'd need to configure some of the following
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
Depending on the selected shape, you need to configure some of the following
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 8]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+properties:
+
+* The :guilabel:`Size type` of the frame, which can be:
+
+  * :guilabel:`Fixed`: using a unique size for all the labels, regardless the
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
  * :guilabel:`Fixed`: using the same size for all the labels, regardless the
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 9]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+    whole text size
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
    size of the text
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 10]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+  * or a :guilabel:`Buffer` over the text's bounding box
+* The :guilabel:`Size` of the frame in X and Y directions, using any
+  :ref:`supported units <unit_selector>`
+* A :guilabel:`Rotation` of the background, between :guilabel:`Sync with label`,
+  :guilabel:`Offset of label` and :guilabel:`Fixed`. The last two require
+  an angle in degrees.
+* An :guilabel:`Offset X,Y` to shift the background item in X and/or Y directions
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
* An :guilabel:`Offset X,Y` to shift the background item in the X and/or Y directions
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 11]
```diff
@@ -177,38 +173,64 @@ text buffer` checkbox in the |labelbuffer| :guilabel:`Buffer` tab. Then you can:
...
+* A :guilabel:`Radius X,Y` to round corners of the background shape (applies
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
* A :guilabel:`Radius X,Y` to round the corners of the background shapes (applies
```

File: source/docs/user_manual/style_library/label_settings.rst (hunk 5 above)
Comment (DelazJ) [reply source/docs/user_manual/style_library/label_settings.rst]: these are three alternatives (url, file, embed) of same level. The way you suggest, it looks like two combinations of levels of two alternatives (from url or file and then being embedded in project or db)

Comment (havatv) [reply source/docs/user_manual/style_library/label_settings.rst]: Four alternatives are listed:
1) From a file
2) From a URL
3) Embedded in the project
4) Embedded in a style database
And I think placing the alternatives in parenthesis emphasises just that.

**Pull request merged**